worker/db_to_csv_worker.go

```diff
@@ -0,0 +1,199 @@
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package worker
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"runtime"
+	"sort"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+	workers "github.com/jrallison/go-workers"
+	"github.com/topfreegames/marathon/model"
+	"github.com/uber-go/zap"
+)
+
+const nameDbToCsv = "db_to_csv_worker"
+
+// ToCSVMessage store th information needed to retrive part of data from the db
+// and save it to the csv
+type ToCSVMessage struct {
+	Query      string
+	Uploader   s3.CreateMultipartUploadOutput
+	PartNumber int
+	Job        model.Job
+	TotalJobs  int
+}
+
+// CompletedPart store information related to one completed part
+type CompletedPart struct {
+	ETag       string
+	PartNumber int64
+}
+
+type completedParts []*s3.CompletedPart
+
+func (a completedParts) Len() int           { return len(a) }
+func (a completedParts) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a completedParts) Less(i, j int) bool { return *a[i].PartNumber < *a[j].PartNumber }
+
+// DbToCsvWorker is the CreateBatchesUsingFiltersWorker struct
+type DbToCsvWorker struct {
+	Logger  zap.Logger
+	Workers *Worker
+}
+
+// NewDbToCsvWorker gets a new DbToCsvWorker
+func NewDbToCsvWorker(workers *Worker) *DbToCsvWorker {
+	b := &DbToCsvWorker{
+		Logger:  workers.Logger.With(zap.String("worker", "DbToCsvWorker")),
+		Workers: workers,
+	}
+	b.Logger.Debug("Configured DbToCsvWorker successfully")
+	return b
+}
+
+func (b *DbToCsvWorker) getPageFromDBWith(message *ToCSVMessage) *[]string {
+	var users []string
+	start := time.Now()
+	_, err := b.Workers.PushDB.DB.Query(&users, message.Query)
+	labels := []string{fmt.Sprintf("game:%s", message.Job.App.Name), fmt.Sprintf("platform:%s", message.Job.Service)}
+	b.Workers.Statsd.Timing("get_page_with_filters", time.Now().Sub(start), labels, 1)
+	checkErr(b.Logger, err)
+	return &users
+}
+
+// return if is the last element or not
+func (b *DbToCsvWorker) redisSaveCompletedPart(message *ToCSVMessage, etag string) bool {
```

Comment:
maybe a better name would be `multiPartUploadCompleted`? the idea here is just to check if all parts where uploaded, right?

Reply:
No, the idea is to save the information of the uploaded part and retrieve the number of uploaded parts.

Reply:
Okay, this is not very explicit. Change the name to `saveCompletedPartToRedis` and add this info before your comment